src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -385,6 +385,54 @@ let CheckNamespaceModuleOrTypeName (g: TcGlobals) (id: Ident) =
     // type names '[]' etc. are used in fslib
     if not g.compilingFSharpCore && id.idText.IndexOfAny IllegalCharactersInTypeAndNamespaceNames <> -1 then 
         errorR(Error(FSComp.SR.tcInvalidNamespaceModuleTypeUnionName(), id.idRange))
+        
+let CheckDuplicatesArgNames (synVal: SynValSig) m =
+    let argNames =
+        synVal.SynInfo.ArgNames
+        |> List.countBy id
+        |> List.filter (fun (_, count) -> count > 1)
+        |> List.map fst
+    for name in argNames do
```

Comment:
synVal.SynInfo.ArgNames |> List.duplicates

(List.duplicates is already present in the compiler codebase)

Reply:
Updated . Thanks